carwings.net/Converters/TimeRequiredJsonConverter.cs

```diff
@@ -23,7 +29,21 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
         public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
         {
-            throw new NotImplementedException();
+            if (value == null)
+            {
+                writer.WriteToken(JsonToken.Null, null);
+                return;
+            }
+
+            TimeSpan duration = (TimeSpan)value;
```

Comment:
this will throw if value is not a timespan

Reply:
Ah.  I'll move this after the if check that throws an ArgumentException if we were passed anything other than a TimeSpan, instead of an InvalidCastException.